test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -214,10 +219,111 @@ def test_one_core_fail(cache_mode):
         # Cache occupancy cannot be lower than before the first core fails and after that
         # should be equal to the sum of occupancy of the first and the remaining core
         cache_occupied_blocks_after = cache.get_occupancy()
-        if cache_occupied_blocks_before > cache_occupied_blocks_after \
-                or cache_occupied_blocks_after != core2.get_occupancy() + core1.get_occupancy():
-            TestRun.fail("Blocks previously occupied by the first core "
-                         "aren't released after this core failure.")
+        if (
+            cache_occupied_blocks_before > cache_occupied_blocks_after
+            or cache_occupied_blocks_after != core2.get_occupancy() + core1.get_occupancy()
+        ):
+            TestRun.fail(
+                "Blocks previously occupied by the first core "
+                "aren't released after this core failure."
+            )
+
+    with TestRun.step("Stop cache."):
+        casadm.stop_all_caches()
+
+    with TestRun.step("Plug back the first core."):
+        core_dev1.plug()
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core1", DiskTypeLowerThan("cache"))
+@pytest.mark.require_disk("core2", DiskTypeLowerThan("cache"))
+def test_one_core_fail_dirty():
```

Comment:
There will be need to solve conflicts but you are right It will be clearer.
I added this new test in PR https://github.com/Open-CAS/open-cas-linux/pull/1346